packages/brain/src/modules/cron/trackValidatorsPerformance/insertPerformanceData.ts

```diff
@@ -33,42 +38,91 @@ export async function insertPerformanceDataNotThrow({
   executionClient: ExecutionClient;
   consensusClient: ConsensusClient;
-  error?: Error;
+  error?: ValidatorPerformanceError;
 }): Promise<void> {
   for (const validatorIndex of activeValidatorsIndexes) {
-    //const liveness = validatorsLiveness.find((liveness) => liveness.index === validatorIndex)?.is_live;
+    if (error) {
+      await insertDataNotThrow({
+        postgresClient,
+        validatorPerformance: {
+          validatorIndex: parseInt(validatorIndex),
+          epoch: currentEpoch,
+          executionClient,
+          consensusClient,
+          error
+        }
+      });
+      continue;
+    }
+
     const attestationsTotalRewards = validatorsAttestationsTotalRewards.find(
       (attestationReward) => attestationReward.validator_index === validatorIndex
     );
-
     if (!attestationsTotalRewards) {
-      logger.error(`${logPrefix}Missing data for validator ${validatorIndex}, att: ${attestationsTotalRewards}`);
+      await insertDataNotThrow({
+        postgresClient,
+        validatorPerformance: {
+          validatorIndex: parseInt(validatorIndex),
+          epoch: currentEpoch,
+          executionClient,
+          consensusClient,
+          error: {
+            code: ValidatorPerformanceErrorCode.MISSING_ATT_DATA,
+            message: `Missing attestation data for validator ${validatorIndex}`
+          }
+        }
+      });
       continue;
     }
 
     const blockProposalStatus = validatorBlockStatusMap.get(validatorIndex);
     if (!blockProposalStatus) {
-      logger.error(
-        `${logPrefix}Missing block proposal data for validator ${validatorIndex}, block: ${blockProposalStatus}`
-      );
+      await insertDataNotThrow({
+        postgresClient,
+        validatorPerformance: {
+          validatorIndex: parseInt(validatorIndex),
+          epoch: currentEpoch,
+          executionClient,
+          consensusClient,
+          error: {
+            code: ValidatorPerformanceErrorCode.MISSING_BLOCK_DATA,
+            message: `Missing block proposal data for validator ${validatorIndex}`
+          }
+        }
+      });
       continue;
     }
 
-    try {
-      logger.debug(`${logPrefix}Inserting performance data for validator ${validatorIndex}`);
-      await postgresClient.insertPerformanceData({
+    await insertDataNotThrow({
+      postgresClient,
+      validatorPerformance: {
         validatorIndex: parseInt(validatorIndex),
         epoch: currentEpoch,
-        blockProposalStatus,
-        attestationsTotalRewards,
-        error: error?.message,
         executionClient,
-        consensusClient
-      });
-      logger.debug(`${logPrefix}Performance data inserted for epoch ${currentEpoch}`);
-    } catch (e) {
-      logger.error(`${logPrefix}Error inserting performance data for validator ${validatorIndex}: ${e}`);
-      continue;
-    }
+        consensusClient,
+        blockProposalStatus,
+        attestationsTotalRewards
+      }
+    });
+  }
```

Comment:
Would be nice to add a debug log that prints something once all data of all vaidators has been written to db

Reply:
already logged in line 124